rules/app_clip.bzl

```diff
@@ -0,0 +1,75 @@
+load("@build_bazel_rules_apple//apple:ios.bzl", rules_apple_ios_app_clip = "ios_app_clip")
+load("//rules:plists.bzl", "process_infoplists")
+load("//rules:force_load_direct_deps.bzl", "force_load_direct_deps")
+load("//rules/internal:framework_middleman.bzl", "dep_middleman", "framework_middleman")
+
+def ios_app_clip(name, infoplists_by_build_setting = {}, **kwargs):
+    """
+    Builds and packages an iOS App Clip.
+
+    The docs for app_clip are at rules_apple
+    https://github.com/bazelbuild/rules_apple/blob/master/doc/rules-ios.md#ios_app_clip
+
+    Args:
+        name: The name of the iOS app clip.
+        infoplists_by_build_setting: A dictionary of infoplists grouped by bazel build setting.
+
+                                     Each value is applied if the respective bazel build setting
+                                     is resolved during the analysis phase.
+
+                                     If '//conditions:default' is not set the value in 'infoplists'
+                                     is set as default.
+        **kwargs: Arguments passed to the ios_app_clip rule as appropriate.
+    """
+
+    deps = kwargs.pop("deps", [])
+    frameworks = kwargs.pop("frameworks", [])
+    testonly = kwargs.pop("testonly", False)
+
+    kwargs["families"] = kwargs.pop("families", ["iphone", "ipad"])
+
+    # Setup force loading here - need to process deps and libs
+    force_load_name = name + ".force_load_direct_deps"
+    force_load_direct_deps(
+        name = force_load_name,
+        deps = deps,
+        tags = ["manual"],
+        testonly = testonly,
+    )
+
+    # Setup framework middlemen - need to process deps and libs
+    fw_name = name + ".framework_middleman"
+    framework_middleman(
+        name = fw_name,
+        framework_deps = deps,
+        tags = ["manual"],
+        testonly = testonly,
+    )
+    frameworks = [fw_name] + frameworks
+
+    dep_name = name + ".dep_middleman"
+    dep_middleman(
+        name = dep_name,
+        deps = deps,
+        tags = ["manual"],
+        testonly = testonly,
+    )
+    deps = [dep_name] + [force_load_name]
+
+    processed_infoplists = process_infoplists(
+        name = name,
+        infoplists = kwargs.pop("infoplists", []),
+        infoplists_by_build_setting = infoplists_by_build_setting,
+        xcconfig = kwargs.pop("xcconfig", {}),
+        xcconfig_by_build_setting = kwargs.pop("xcconfig_by_build_setting", {}),
```

Comment:
Oh I see, yep I think the same should apply there if that's the case, cc @mattrobmattrob let me know if this makes sense as I think you introduced that!

Reply:
And I was but a forwarder of this implementation for the extension use-case:

https://github.com/bazel-ios/rules_ios/blob/1ce3284caf2bf1d062b149efeab183d4d14d31f7/rules/app.bzl#L104

Might be nice to get a bit of unification across these 3 extension/app implementation to share the middleman creation since AFAIK it's the same.

Reply:
+1, IMO can also be done in a follow up to not scope creep this PR if you prefer @luispadron

Reply:
Yup, also very okay with a follow up. Would be some nice clean up though. 👏